lib/logstash-docket/artifact_plugin.rb

```diff
@@ -118,6 +119,20 @@ def with_embedded_plugins
       end
     end
 
+    ##
+    # Returns an {@link Enumerable[Plugin]} containing itself and any wrapped plugins
+    #
+    # @return [Enumerable[Plugin]]
+    def with_wrapped_plugins(alias_mappings)
```

Comment:
Giving a proper name for this method was hard :)
As my understanding, we have `integration` & `alias` types of plugins which are wrappers for artifact plugins.